pkg/kubelet/volume/volume_manager.go

```diff
@@ -0,0 +1,357 @@
+
+	// operationExecutor is used to start asynchronous attach, detach, mount,
+	// and unmount operations.
+	operationExecutor operationexecutor.OperationExecutor
+
+	// reconciler runs an asynchronous periodic loop to reconcile the
+	// desiredStateOfWorld with the actualStateOfWorld by triggering attach,
+	// detach, mount, and unmount operations using the operationExecutor.
+	reconciler reconciler.Reconciler
+
+	// desiredStateOfWorldPopulator  runs an asynchronous periodic loop to
+	// populate the desiredStateOfWorld using the kublet PodManager.
+	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+}
+
+func (vm *volumeManager) Run(stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	glog.Infof("Starting Kubelet Volume Manager")
+
+	go vm.reconciler.Run(stopCh)
+	go vm.desiredStateOfWorldPopulator.Run(stopCh)
+
+	<-stopCh
+	glog.Infof("Shutting down Kubelet Volume Manager")
+}
+
+func (vm *volumeManager) GetVolumesForPod(
+	podName types.UniquePodName) container.VolumeMap {
+	return vm.getVolumesForPodHelper(podName, nil /* pod */)
+}
+
+func (vm *volumeManager) GetVolumesForPodAndAppendSupplementalGroups(
+	pod *api.Pod) container.VolumeMap {
+	return vm.getVolumesForPodHelper("" /* podName */, pod)
+}
+
+func (vm *volumeManager) GetVolumesInUse() []api.UniqueVolumeName {
+	attachedVolumes := vm.actualStateOfWorld.GetAttachedVolumes()
+	volumesInUse :=
+		make([]api.UniqueVolumeName, 0 /* len */, len(attachedVolumes) /* cap */)
+	for _, attachedVolume := range attachedVolumes {
+		if attachedVolume.PluginIsAttachable {
+			volumesInUse = append(volumesInUse, attachedVolume.VolumeName)
+		}
+	}
+
+	return volumesInUse
+}
+
+// getVolumesForPodHelper is a helper method implments the common logic for
+// the GetVolumesForPod methods.
+func (vm *volumeManager) getVolumesForPodHelper(
+	podName types.UniquePodName, pod *api.Pod) container.VolumeMap {
+	if pod != nil {
+		podName = volumehelper.GetUniquePodName(pod)
+	}
+	podVolumes := make(container.VolumeMap)
+	for _, mountedVolume := range vm.actualStateOfWorld.GetMountedVolumesForPod(podName) {
+		podVolumes[mountedVolume.OuterVolumeSpecName] =
+			container.VolumeInfo{Mounter: mountedVolume.Mounter}
+		if pod != nil {
+			err := applyPersistentVolumeAnnotations(
+				mountedVolume.VolumeGidValue, pod)
+			if err != nil {
+				glog.Errorf("applyPersistentVolumeAnnotations failed for pod %q volume %q with: %v",
+					podName,
+					mountedVolume.VolumeName,
+					err)
+			}
+		}
+	}
+	return podVolumes
+}
+
+func (vm *volumeManager) WaitForAttachAndMount(pod *api.Pod) error {
+	expectedVolumes := getExpectedVolumes(pod)
+	if len(expectedVolumes) == 0 {
+		// No volumes to verify
+		return nil
+	}
+
+	glog.Infof("Waiting for volumes to attach and mount for pod %q", format.Pod(pod)) // TODO drop to V(3)?.
+	uniquePodName := volumehelper.GetUniquePodName(pod)
+
+	// If pod is already mounted, force remount to reflect any changes in the
```

Comment:
Okay, so here's what I would say:

`Some pods expect to have Setup called over and over again to update.  Remount plugins for which this is true.`